connector/google/google.go

```diff
@@ -296,3 +310,18 @@ func createDirectoryService(serviceAccountFilePath string, email string) (*admin
 	}
 	return srv, nil
 }
+
+// uniqueGroups returns the unique groups of a slice
+func uniqueGroups(groups []string) []string {
+	keys := make(map[string]bool)
+	unique := []string{}
+
+	for _, group := range groups {
+		if _, value := keys[group]; !value {
+			keys[group] = true
+			unique = append(unique, group)
+		}
+	}
+
+	return unique
+}
```

Comment:
If you use a `struct`, it's technically more efficient, I would do this as (this is very much a nit)
```suggestion
	keys := make(map[string]struct{})
	unique := []string{}

	for _, group := range groups {
		if _, exists := keys[group]; !exists {
			keys[group] = struct{}{}
			unique = append(unique, group)
		}
	}

	return unique
}
```